src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/pre_compute.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+
+namespace ov :: npuw :: patterns :: pre_compute {
+
+class RopePatternDesc {
+protected:
+    ov::matcher_pass_callback verifier_cb;
+public:
+    std::shared_ptr<ov::Node> pattern;
+    std::shared_ptr<ov::Node> matched_inv_freq;
+    std::shared_ptr<ov::Node> matched_position_ids;
+    std::shared_ptr<ov::Node> matched_sin;
+    std::shared_ptr<ov::Node> matched_cos;
+    std::shared_ptr<ov::Node> matched_concat;
+    //TODO: can be moved to private and init in ctor
+    ov::matcher_pass_callback callback;
+
+    ov::matcher_pass_callback make_matcher_callback() {
+        return [this](pass::pattern::Matcher& m) {
+            if (!verifier_cb(m)) {
+                return false;
+            }
```

Comment:
I thought transformations support conditionals, @itikhono can you suggest a better alternative?